reference.py

```diff
@@ -56,7 +56,7 @@ def get_pubkey_from_input(vin: VinInfo) -> ECPubKey:
             if (pubkey.valid) & (pubkey.compressed):
                 return pubkey
     if is_p2wpkh(vin.prevout):
-        txin = CTxInWitness().deserialize(vin.txinwitness)
```

Comment:
why remove this line? the `serialize_witness_stack` function turns the stack into a hex encoded byte vector and `CTxInWitness().deserialize()` takes the hex encoding and reads it into a stack.

Reply:
I think removing this should break all the tests, did you try running `python3 reference.py send_and_receive_test_vectors.json` ?

Reply:
> I think removing this should break all the tests, did you try running `python3 reference.py send_and_receive_test_vectors.json` ?

Yeah, I did. I had to remove that line because vin.txinwitness has already been deserialized and it's a `CTxInWitness` object. See https://github.com/S3RK/bip-352/blob/04b1d3ae9ea38c587ad3bb780dc0a1ea2fb81c66/reference.py#L270

Reply:
Ah, makes sense! I was really confused. Not sure how that was working before.